draft-ietf-quic-transport.md

```diff
@@ -716,17 +716,28 @@ sequence of frames, as described in {{frames}}.
 A sender can coalesce multiple QUIC packets (typically a Cryptographic Handshake
 packet and a Protected packet) into one UDP datagram.  This can reduce the
 number of UDP datagrams needed to send application data during the handshake and
-immediately afterwards.  A packet with a short header does not include a length,
-so it has to be the last packet included in a UDP datagram.
+immediately afterwards.
 
-The sender MUST NOT coalesce QUIC packets belonging to different QUIC
-connections into a single UDP datagram.
+Senders SHOULD coalesce packets in order of increasing encryption levels
+(Initial, Handshake, 0-RTT, 1-RTT).  A packet with a short header does not
+include a length, so it will always be the last packet included in a UDP
+datagram.
+
+Senders MUST NOT coalesce QUIC packets with different Destination Connection
+IDs into a single UDP datagram. Receivers SHOULD ignore any subsequent packets
+with a different Destination Connection ID than the first packet in the
+datagram.
 
 Every QUIC packet that is coalesced into a single UDP datagram is separate and
 complete.  Though the values of some fields in the packet header might be
 redundant, no fields are omitted.  The receiver of coalesced QUIC packets MUST
 individually process each QUIC packet and separately acknowledge them, as if
-they were received as the payload of different UDP datagrams.
+they were received as the payload of different UDP datagrams.  If one or more
+packets in a datagram cannot be processed, either because the keys are not yet
+available or because decryption fails, the receiver MUST still attempt to
```

Comment:
It might make sense to add the case where the long header type is invalid. Explicitly allowing a sender to emit an invalid long header type will help us grease the first octet.

Reply:
I was initially dubious about this, thinking that the Length field is type-specific.  But it appears to only be version-specific, which means you can still trust the Length on an unknown type.

Done, though the wording feels a little awkward; suggestions for improved flow welcome.